vdev/src/testing/runner.rs

```diff
@@ -47,6 +47,29 @@ fn detect_container_tool() -> OsString {
     fatal!("No container tool could be detected.");
 }
 
+fn get_rust_version() -> String {
+    match RustToolchainConfig::parse() {
+        Ok(config) => config.channel,
+        Err(error) => fatal!("Could not read `rust-toolchain.toml` file: {error}"),
+    }
+}
+
+/// True if a docker image exists that has all of the integration tests
+/// features compiled.
+pub fn full_runner_available() -> Result<bool> {
+    let mut command = dockercmd([
+        "image",
+        "ls",
+        &format!("vector-test-runner-{}", get_rust_version()),
+    ]);
+
+    // $ docker image ls vector-test-runner-1.69.0
+    // REPOSITORY                  TAG       IMAGE ID       CREATED       SIZE
+    // vector-test-runner-1.69.0   latest    7e5e70a96428   13 days ago   1.76GB
+
+    Ok(command.check_output()?.lines().count() == 2)
+}
```

Comment:
How does it know if the image is up-to-date?

Reply:
In a later step, it runs the command to build it, so if there is any discrepancy in the build steps, it would re-compile the pieces that are out of date. If it's all up to date it just uses the cached state.

Reply:
teeechnically if someone were to run with the new -a flag , then go and change a bunch of stuff or checkout a branch that changed every component , then ran the int tests again, whether with -a or not, it would have to re compile for those changes across the all int test features (assuming they didn't prune the docker image).

I imagine though that developers are less likely to use this option locally.